spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaProducerProperties.java

```diff
@@ -41,6 +43,14 @@ public void setBufferSize(int bufferSize) {
 		this.bufferSize = bufferSize;
 	}
 
+	public int getMaxRequestSize() {
```

Comment:
For the sake of the previous example: bufferSize is set here https://github.com/spring-cloud/spring-cloud-stream/blob/1.0.x/spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java#L312

However, since ProducerMetadata doesn't have a field for it, it will need to be set the `additional` set of properties, eg.

https://github.com/spring-cloud/spring-cloud-stream/blob/1.0.x/spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java#L314